### PR TITLE
Bluetooth: Mesh: Add support for MshPRTd1.1r18, MshMBT_d1.0r04_PRr00, MshDFU_d1.0r04_PRr00 specs

### PR DESCRIPTION
**Pull request by PavelVPV**
This PR adds support for MshPRTd1.1r18, MshMBT_d1.0r04_PRr00, MshDFU_d1.0r04_PRr00 draft specifications. The details are present below.

### New features

From the Mesh Protocol d1.1r18 specification (https://www.bluetooth.com/specifications/specs/mesh-protocol-1-1/):
- Enhanced Provisioning Authentication support
- Mesh Remote Provisioning support including:
  - Remote Provisioning Server and Client models
  - Composition Data Page 128 and Models Metadata Page 128 support
- Large Composition Data support including:
  - Large Composition Data Server and Client models
  - Models Metadata Page 0 support
- New Transport Segmentation and Reassembly (SAR) implementation including:
  - SAR Configuration Server and Client models
- Mesh Private Beacons support including:
  - Mesh Private Beacon Server and Client models
- Opcodes Aggregator support including:
  - Opcodes Aggregator Server and Client models
- Proxy Solicitation support including:
  - Solicitation PDU RPL Configuration Server and Client models
  - On-Demand Private Proxy Server and Client models
- Composition Data Page 1 support
- Other Mesh Profile Enhancements

From the Mesh Binary Large Object Transfer Model d1.0r04_PRr00 specification (https://www.bluetooth.com/specifications/specs/mesh-binary-large-object-transfer-model-2/):
- BLOB Transfer Server & Client models

From the Mesh Device Firmware Update Model d1.0r04_PRr00 specification (https://www.bluetooth.com/specifications/specs/mesh-device-firmware-update-model-2/):
- Firmware Update Server and Client models
- Firmware Distribution Server model

#### Compatibility with the Mesh Profile 1.0.1 implementation

The new features (including new transport SAR implementation) are enabled by the `CONFIG_BT_MESH_V1d1` Kconfig option thus not affecting qualification of the Mesh Profile 1.0.1 implementation.

#### Shell support

This PR adds shell support for the all features listed above.
The new models support is added to the `mesh_shell` application.

#### Documentation

The documentation is present for all features listed above except:
- SAR Configuration models
- Large Composition Data models
- Remote Provisioning

#### Current tests

All currently implemented BabbleSim tests run for both the Mesh Profile 1.0.1 and the Mesh Protocol d1.1r18 implementations.

#### New tests

This PR implements BabbleSim tests for the following features:
- Mesh Remote Provisioning
- Large Composition Data
- Opcodes Aggregator
- Transport SAR Configuration
- BLOB Transfer Server & Client models
- Firmware Update Server and Client models
- Firmware Distribution Server model

### CODE REVIEW

**Comment by LingaoM** [comment]
Done this feature public now?

**Comment by PavelVPV** [comment]
> Done this feature public now?

The specs are publicly available here: https://www.bluetooth.com/specifications/specs/?types=specification&keyword=mesh&filter=

File: tests/bluetooth/mesh_shell/testcase.yaml [hunk 1]
```diff
@@ -3,9 +3,7 @@ common:
   harness: keyboard
 tests:
   bluetooth.mesh.mesh_shell:
-    platform_allow: qemu_x86 nrf51dk_nrf51422
-    integration_platforms:
-      - qemu_x86
+    platform_allow: qemu_cortex_m3 qemu_x86 nrf51dk_nrf51422 nrf52dk_nrf52832
```
Comment (alxelax) [inline tests/bluetooth/mesh_shell/testcase.yaml:6]: Seems this unnecessary change sneaked somehow. The initial upstream file is more correct. I guess changes in this file should be reverted.

**Comment by Thalley** [comment]
@PavelVPV Are those specification public? Zephyr shouldn't contain code for non-public specification IMO

**Comment by PavelVPV** [comment]
> https://www.bluetooth.com/specifications/specs/?types=specification&keyword=mesh&filter=

Yes, they are public. Anyone can download them here: https://www.bluetooth.com/specifications/specs/?types=specification&keyword=mesh&filter=
Though the specs are in draft state yet.

**Comment by LingaoM** [comment]
> > @PavelVPV Are those specification public? Zephyr shouldn't contain code for non-public specification IMO
> 
> 
> 
> Yes, they are public. Anyone can download them here: https://www.bluetooth.com/specifications/specs/?types=specification&keyword=mesh&filter=
> 
> Though the specs are in draft state yet.

It's best to put this information in the initial description so it's self-explanatory.

**Comment by PavelVPV** [comment]
> It's best to put this information in the initial description so it's self-explanatory.

Links are added

**Comment by jhedberg** [comment]
Lines added/removed: `+32,564 −809`. That's one of the biggest single pull requests I've seen in awhile, and will make the review process quite long and cumbersome. Any chance you could split this up into smaller, easier to review, parts, e.g. one PR per new feature?

**Comment by PavelVPV** [comment]
> Lines added/removed: `+32,564 −809`. That's one of the biggest single pull requests I've seen in awhile, and will make the review process quite long and cumbersome. Any chance you could split this up into smaller, easier to review, parts, e.g. one PR per new feature?

We'd like to not split this PR into multiple ones. But the changes are split in commits per each feature. Shell commands for each feature are in separate commits. And BabbleSim tests are split in separate commits.

**Comment by Balaklaka** [comment]
The features that has been added should be added to release notes.

**Review by jhedberg — COMMENTED** [review]
Some minor things spotted after a *very* quick read-through. Not sure they're severe enough to block the PR for, but definitely things that should be eventually fixed.

**Comment by Balaklaka** [comment]
Html doc can be found there: https://builds.zephyrproject.io/zephyr/pr/55393/docs/connectivity/bluetooth/api/mesh/models.html

**Review by jhedberg — DISMISSED** [review]

**Review by Balaklaka — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by LingaoM — APPROVED** [review]

**Review by alxelax — APPROVED** [review]
